src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -911,7 +911,7 @@ protected override Task<bool> ReadAtDeltaDeletedLinkImplementationAsync()
                 ? null
                 : this.jsonLightResourceDeserializer.ContextUriParseResult.SelectQueryOption;
 
-            SelectedPropertiesNode selectedProperties = SelectedPropertiesNode.Create(selectQueryOption, this.CurrentResourceType, this.jsonLightInputContext.Model);
+            SelectedPropertiesNode selectedProperties = SelectedPropertiesNode.Create(selectQueryOption, (this.CurrentResourceTypeReference !=null) ? this.CurrentResourceTypeReference.AsStructured().StructuredDefinition() : null, this.jsonLightInputContext.Model);

```

Comment:
>(this.CurrentResourceTypeReference !=null) [](start = 105, length = 42)

So we are asserting that, if the CurrentResourceTypeReference is non-null, then it is a structured type, right? Seems reasonable.